### PR TITLE
Fix how manpages are generated

### PR DESCRIPTION
**Pull request by cmouse**

### CODE REVIEW

File: docs/core/pigeonhole-man/index.md
Comment (slusarz) [inline docs/core/pigeonhole-man/index.md:1]: It was an explicit decision to merge pigeonhole and core documentation in the new system.

Why do we need to distinguish between the two?  It's not a separate project - it's integrated into Dovecot development.  To most users, it just looks like a separate package, but no other dovecot package separates their documentation and brands their functionality.  No other plugin does this either.

What benefit does separating the two have for a user, other than making the documentation more complicated?  A user not installing pigeonhole is no different than a user not installing/activating a plugin.

File: lib/utility.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ export async function manFiles() {
 		}
 	}
 
-	return [ 'docs/core/man/*.[[:digit:]].md' ]
+	return [ join('docs',component,'*.[[:digit:]].md') ]
```
Comment (slusarz) [inline lib/utility.js:84]: Again, this pigehonhole stuff is adding complexity to build-scripts and the like, while not providing any value to a user.

A user cares about "sieve commands".  It doesn't care that these are "pigeonhole sieve commands".  In fact, this is more confusing: a user now has to learn about the difference between pigeonhole and dovecot to understand where to find the documentation.  And what is that difference?  Does a user care?

**Review by slusarz — CHANGES_REQUESTED** [review]

**Review by slusarz — CHANGES_REQUESTED** [review]
There is also another reference to "pandoc" at line 3 of "generate_man.js" that should be reviewed (GH reviews do not allow you to mark lines for review outside of the diff scope).

File: util/generate_man.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ import matter from 'gray-matter'
 import pdc from 'pdc'
 import path from 'path'
 import { manFiles } from '../lib/utility.js'
+import remarkMan from 'remark-man'
+import remarkParse from 'remark-parse'
+import remarkGfm from 'remark-gfm'
+import {unified} from 'unified'
```
Comment (slusarz) [inline util/generate_man.js:20]: Should be '{ unified }' for coding consistency.

File: util/generate_man.js [hunk 3]
```diff
@@ -31,75 +36,75 @@ program
...
+		const fparts = path.basename(f).split('.')
+		const result = await unified().
+			use(remarkParse).
+			use(processDovecotMd).
+			use(remarkGfm).
+			use(remarkMan, {
+				manual: 'Dovecot',
+				section: parseInt(fparts[1]),
+				version: gitHash,
+				name: fparts[0].toUpperCase()
```
Comment (slusarz) [inline util/generate_man.js:117]: The name header is duplicated in the output.

i.e., man ./doveadm-who.1:

```
DOVEADM-WHO(1)                      Dovecot                     DOVEADM-WHO(1)

NAME
       DOVEADM-WHO - doveadm-who

NAME
       doveadm-who - Show who is logged into the Dovecot server

SYNOPSIS
```

**Review by slusarz — CHANGES_REQUESTED** [review]

**Comment by slusarz** [comment]
If we are looking for a way to only generate man pages for, e.g., core and 'pigeonhole' (for packaging reasons) - which is a valid use-case - the preferred way is to get rid of all this extra complexity introduced by this MR and simply add an additional entry to the man markdown's frontmatter identifying which component the man file goes with.  So, e.g.:

```
---
layout: doc
title: pigeonhole
dovecotComponent: pigeonhole
---

[markdown content...]
```

We are already parsing the file with gray-matter in the script, so a single line: `if (parsed.data.dovecotComponent == 'pigeonhole')` is all we need.

File: lib/utility.js (hunk 1 above)
Comment (cmouse) [reply lib/utility.js:84]: Our own build system cares though as dovecot and pigeonhole are separate. I need some way to ship manpages by component.

File: util/generate_man.js (hunk 3 above)
Comment (cmouse) [reply util/generate_man.js:112]: that's because the file is wrong. i didn't yet fix all the manpages.

Comment (cmouse) [reply util/generate_man.js:112]: oh right.

Comment (slusarz) [reply util/generate_man.js:112]: It does seem to want you to not include a Name header in the markdown code, and instead use the argument parameter.

Seems like it requires the page to look like:

```
# doveadm-who(1) -- Show who is logged into the Dovecot server
```

So the documentation pages would no longer have "NAME" block.

Comment (cmouse) [reply util/generate_man.js:112]: this seems to be correct behavior by man

**Pull request merged**